src/ARAnchorManager.js

```diff
@@ -0,0 +1,93 @@
+// import { EventEmitter } from 'events';
+import { Object3D, Vector3, Matrix4 } from 'three';
+/**
+ * A class to manage registration and unregistration of anchors
+ * for any THREE.Object3D. Anchors are world space and handle updating
+ * the position of the object to which one is assigned.
+ * AnchorManager emits events pertaining to anchors.
+ */
+export default class ARAnchorManager /* extends EventEmitter */ {
+  constructor(vrDisplay) {
+    // super();
+    if (!(vrDisplay instanceof VRDisplay) || !vrDisplay.getAnchors) {
+      throw new Error("invalid vr display parameter");
+    }
+    this.vrDisplay_ = vrDisplay;
+    this.anchorsToObject3Ds_ = new Map();
+    this.object3DsToAnchors_ = new Map();
+    this.vrDisplay_.addEventListener('anchorsupdated', 
+        event => this.onAnchorsUpdated_(event));
+    // This attribute is used to get the scale value from the anchor model
+    // matrix but not use it. It is just to avoid garbage collection.
+    this.scale_  = new Vector3();
+    // This attribute is used to calculate the matrix to be passed to the
+    // anchor creation code only using the position and quaternion from the
+    // object3d.
+    this.matrix_ = new Matrix4();
+    return this;
+  }
+  /**
+   * Assign a new anchor to an THREE.Object3D.
+   */
+  addAnchorForObject3D(object3d) {
+    if (!(object3d instanceof Object3D)) {
+      throw new Error('invalid object3d trying to add an anchor');
+    };
+    // Do not allow to create an anchor for an object3d that already has one.
+    if (this.object3DsToAnchors_.get(object3d)) {
```

Comment:
Improved it in a different way. The add method will return the actual VRAnchor instance (in case the caller wants to know about them. The "anchorsupdated" event will also pass both the object3ds and the anchors.